flink-tests/src/test/java/org/apache/flink/test/javaApiOperators/CustomDistributionITCase.java

```diff
@@ -175,6 +175,77 @@ else if (pIdx > 0 && pIdx < dist.getParallelism() - 1) {
 		env.execute();
 	}
 
+	@Test
+	public void testPartitionKeyLessDistribution() throws Exception{
+		/*
+		 * Test the number of keys less than the number of distribution fields
+		 */
+
+		ExecutionEnvironment env = ExecutionEnvironment.createLocalEnvironment();
+
+		DataSet<Tuple3<Integer, Long, String>> input1 = CollectionDataSets.get3TupleDataSet(env);
+		final TestDataDist2 dist = new TestDataDist2();
+
+		env.setParallelism(dist.getParallelism());
+
+		DataSet<Boolean> result = DataSetUtils
+				.partitionByRange(input1, dist, 0)
+				.mapPartition(new RichMapPartitionFunction<Tuple3<Integer, Long, String>, Boolean>() {
+
+								  @Override
+								  public void mapPartition(Iterable<Tuple3<Integer, Long, String>> values, Collector<Boolean> out) throws Exception {
+									  int pIdx = getRuntimeContext().getIndexOfThisSubtask();
+
+									  for (Tuple3<Integer, Long, String> s : values) {
+										  boolean correctlyPartitioned = true;
+										  if (pIdx == 0) {
+											  Integer[] upper = dist.boundaries[0];
+											  if (s.f0.compareTo(upper[0]) > 0) {
+												  correctlyPartitioned = false;
+											  }
+										  } else if (pIdx > 0 && pIdx < dist.getParallelism() - 1) {
+											  Integer[] lower = dist.boundaries[pIdx - 1];
+											  Integer[] upper = dist.boundaries[pIdx];
+											  if (s.f0.compareTo(upper[0]) > 0 || (s.f0.compareTo(lower[0]) <= 0)) {
+												  correctlyPartitioned = false;
+											  }
+										  } else {
+											  Integer[] lower = dist.boundaries[pIdx - 1];
+											  if ((s.f0.compareTo(lower[0]) <= 0)) {
+												  correctlyPartitioned = false;
+											  }
+										  }
+
+										  if (!correctlyPartitioned) {
+											  fail("Record was not correctly partitioned: " + s.toString());
+										  }
+									  }
+								  }
+							  }
+				);
+
+		result.output(new DiscardingOutputFormat<Boolean>());
+		env.execute();
+	}
+
+	@Test(expected = IllegalArgumentException.class)
+	public void testPartitionMoreThanDistribution() throws Exception{
+		/*
+		 * Test the number of keys larger than the number of distribution fields
+		 */
+
+		ExecutionEnvironment env = ExecutionEnvironment.createLocalEnvironment();
+
+		DataSet<Tuple3<Integer, Long, String>> input1 = CollectionDataSets.get3TupleDataSet(env);
+		final TestDataDist2 dist = new TestDataDist2();
+
+		DataSet<Tuple3<Integer, Long, String>> result = DataSetUtils
+				.partitionByRange(input1, dist, 0, 1, 2);
+
+		result.output(new DiscardingOutputFormat<Tuple3<Integer, Long, String>>());
```

Comment:
I think you can remove this code. The test should fail at `partitionByRange(...)`, right?